python/peak_routing_png_image.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:light
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.12.0
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# # Peak routing
+
+# _Contributed by Dr. Stephan Fuchs_
+#
+# In this tutorial we will:
+# - extract some mountain peaks from OpenStreetMap using the `overpy` package
+# - calculate the routes to the peaks from different starting points
+# - save the maps as PNG files using an intermediate HTML representation
+# as well as the `selenium` and `webdriver-manager` packages
+
+# + [markdown] pycharm={"name": "#%% md\n"}
+# #### Create an output folder for the maps if it doesn't exist already
+
+# + pycharm={"name": "#%%\n"}
+import os
+
+map_folder = f"{os.getcwd()}/maps"
+
+if not os.path.exists(map_folder):
+    print("Creating maps folder")
+    os.makedirs(map_folder)
+
+# + [markdown] pycharm={"name": "#%% md\n"}
+# #### Get the coordinates of the peaks
+# You can directly download the peak data via the overpy package.
+# It uses the same query language as [overpass-turbo](https://overpass-turbo.eu/#).
+#
+# Here the peaks where the name matches one of the `peak_names` will be extracted from the given bounding box.
+
+# + pycharm={"name": "#%%\n"}
+import overpy
+import json
+
+api = overpy.Overpass()
+
+bbox = [50.82263785103416, 13.996753692626951, 50.843670947425615, 14.116744995117186]
+peak_names = ["Zeisigstein", "Herkuleskopf", "Kristin Hrádek"]
+
+# Query natural peaks from overpass api within bounding box
+result = api.query(
+    f"""node["natural"="peak"][name~"^({"|".join(peak_names)})$"]({",".join([str(c) for c in bbox])});out;""")
```

Comment:
1. valid point.. guess i alt-enterd for auto completion, single quote works as well.
2. that's the overpass query language for you, writing a whole wrapper for that would be out of scope for an example i guess..

what about the comment right above it that tells that it queries natural peaks from overpass api within the bounding box, you think that's not sufficient?
there is not much more info i can add except for saying that it will query point objects where the name matches exactly one of the strings

Reply:
yeah,  that's the comment in natural language.
I was thinking along the lines of having the query there without all the f"{}" and python-functions in there, so people familiar with overpass can read it or copy it into overpass to understand?

Reply:
Ah i see. I was thinking in the direction of making the values easily interchangable if one wants to run the example on other bbox & peaks.
maybe a print of the query before sending it to make it possible to copy from the cell output?

Reply:
Did a print of the output to easily see the whole query

Reply:
Nice, I like the approach and the usability :)